### DISCUSSION TITLE
Rendering in a Remix app

### DISCUSSION THREAD

**Discussion by khera** [scalar/scalar#1504] [Q&A]
I tried to use the React example to make a page in my Remix app to render my spec file. I'm stuck at this error:

```
6:00:41 PM [vite] Internal server error: Unknown file extension ".css" for node_modules/@scalar/api-reference-react/dist/index.css
```

According to the Remix + Vite docs, the Vite builder should just work with css file imports, and they offer nothing in the way of any configurations I should set.

Does anyone have a full recipe to render my API spec with Remix?

**Comment by amritk** [comment]
Hey @khera, I haven't tried remix yet but we should have an integration that might work on it soon. In the meantime do you think you could setup a minimal reproducible example? 

We have css included in our js which vite can handle. Usually its a plugin that can't handle it. It was just a single config option to get it working in nuxt, so if you set up the example maybe I can figure out what it is supposed to be in remix.

**Comment by khera** [comment]
While making a test case, I found https://github.com/vitejs/vite/issues/9487 which says to add this to my `vite.config.js` file:

```ts
ssr: {
  noExternal: ['@scalar/api-reference-react']
},
```

and it worked. For reference, the entire page I made is this simple:

```tsx
import { ApiReferenceReact } from '@scalar/api-reference-react'
import apiSpec from '~/petstore.json';

export default function ApiSpec() {
  return (
    <>
      <ApiReferenceReact
        configuration={{
          spec: {
            content: apiSpec
          },
        }}
      />
    </>
  );
}
```

My only question left is if there is a way to reduce the number of packages this needs. It installed 627 packages to add `@scalar/api-reference-react`.

**Reply by amritk** [reply]
yea currently there's a lot of packages required for:
- displaying the markdown
- live code highlighting

we do have plans to reduce this, but just not at this moment.

**Reply by amritk** [reply]
were you using the references internally to develop your api, or externally for end users? If the second case we will build a next.js integration soon which will do most of the work on the server, so the end user will get served a much lighter bundle

**Reply by khera** [reply]
I'm using it for public reference.

**Reply by amritk** [reply]
Yea then the next integration will help, you can sub to the tracking ticket here:

https://github.com/scalar/scalar/issues/1565
